hail/python/hail/methods/qc.py

```diff
@@ -492,13 +512,548 @@ def n_discordant(counter):
+    VEP executable. The inputs are `consequence` and `tolerate_parse_error` which are user-defined parameters to :func:`.vep`,
+    `part_id` which is the partition ID, `input_file` which is the path to the input file where the input data can be found, and
+    `output_file` is the path to the output file where the VEP annotations are written to. An example is shown below:
+
+    .. code-block:: python3
+
+        def vep_command(self,
+                        consequence: bool,
+                        tolerate_parse_error: bool,
+                        part_id: int,
+                        input_file: Optional[str],
+                        output_file: str) -> List[str]:
+            vcf_or_json = '--vcf' if consequence else '--json'
+            input_file = f'--input_file {input_file}' if input_file else ''
+            return f'''/vep/vep {input_file} \
+        --format vcf \
+        {vcf_or_json} \
+        --everything \
+        --allele_number \
+        --no_stats \
+        --cache \
+        --offline \
+        --minimal \
+        --assembly GRCh37 \
+        --dir={self.data_mount} \
+        --plugin LoF,human_ancestor_fa:{self.data_mount}/loftee_data/human_ancestor.fa.gz,filter_position:0.05,min_intron_size:15,conservation_file:{self.data_mount}/loftee_data/phylocsf_gerp.sql,gerp_file:{self.data_mount}/loftee_data/GERP_scores.final.sorted.txt.gz \
+        -o STDOUT
+        '''
+
+    The following environment variables are added to the job's environment:
+
+     - `VEP_BLOCK_SIZE` - The maximum number of variants provided as input to each invocation of VEP.
+     - `VEP_PART_ID` - Partition ID.
+     - `VEP_DATA_MOUNT` - Location where the vep data is mounted (same as `data_mount` in the config).
+     - `VEP_CONSEQUENCE` - Integer equal to 0 or 1 on whether `csq` is False or True.
+     - `VEP_TOLERATE_PARSE_ERROR` - Integer equal to 0 or 1 on whether `tolerate_parse_error` is False or True.
+     - `VEP_OUTPUT_FILE` - String specifying the local path where the output TSV file with the VEP result should be located.
+     - `VEP_INPUT_FILE` - String specifying the local path where the input VCF shard is located for all jobs.
+
+    The `VEP_INPUT_FILE` environment variable is not available for the single job that computes the consequence header when
+    ``csq=True``
+    """
+
+    vep_json_typ: hl.expr.HailType
+    data_bucket: str
+    data_mount: str
+    regions: List[str]
+    image: str
+    env: Dict[str, str]
+    reference_data_is_bucket_requester_pays: bool
+    cloud: str
+    batch_run_command: List[str]
+    batch_run_csq_header_command: List[str]
+
+    @abc.abstractmethod
+    def vep_command(self,
+                    consequence: bool,
+                    tolerate_parse_error: bool,
+                    part_id: int,
+                    input_file: Optional[str],
+                    output_file: str) -> List[str]:
+        raise NotImplementedError
+
+
+class HailVEPConfig(BaseVEPConfig, abc.ABC):
```

Comment:
I'm worried folks will find it confusing that there's a thing called `BaseVEPConfig` and a thing called `HailVEPConfig` and when configuring VEP for Hail they're *not* supposed to use the thing called `HailVEPConfig`.

Why not make `HailVEPConfig` the base class (and just not have `BaseVEPConfig`)? Then you can use the standard constructor documentation to describe each argument/field and their type. You can still leave `vep_command` as an abstract method.

Is `from_dict` used at all?

Reply:
Going this route, you'd need to make the `vep_json_type` something like `GRCh37_JSON_TYPE`. This also has the advantage of exposing `vep_json_typ` through a public class rather than as a field on an undocumented and unexported class. That helps a user discover it and use it as an example from which to define their own JSON type.

Reply:
Or you could push GRCh37_JSON_TYPE down into the GRCh37 class. That also seems fine.